src/Symfony/Bundle/FrameworkBundle/Translation/PhpExtractor.php

```diff
@@ -11,250 +11,11 @@
 
 namespace Symfony\Bundle\FrameworkBundle\Translation;
 
-use Symfony\Component\Finder\Finder;
-use Symfony\Component\Translation\Extractor\AbstractFileExtractor;
-use Symfony\Component\Translation\MessageCatalogue;
-use Symfony\Component\Translation\Extractor\ExtractorInterface;
+@trigger_error(sprintf('The class "%s" is deprecated since version 3.4 and will be removed in 4.0. Use "%s" instead. ', PhpExtractor::class, \Symfony\Component\Translation\Extractor\PhpExtractor::class), E_USER_DEPRECATED);
 
 /**
- * PhpExtractor extracts translation messages from a PHP template.
- *
- * @author Michel Salib <michelsalib@hotmail.com>
+ * @deprecated since version 3.4 and will be removed in 4.0. Use Symfony\Component\Translation\Extractor\PhpExtractor instead
  */
-class PhpExtractor extends AbstractFileExtractor implements ExtractorInterface
+class PhpExtractor extends \Symfony\Component\Translation\Extractor\PhpExtractor
```

Comment:
this doesn't work as a BC layer: `PhpExtractor instanceof PhpExtractor` should be true whatever the namespaces.
the proper way to do so is using class_alias+class_exists.
see BC layer of ChildDefinion/DefinitionDecorator as an example.

Reply:
Well, the issue with class_alias is that you won't get a deprecation warning when using the old class name if the new one was autoloaded first.

For PhpExtractor, support for `ComponentExtractor instanceof BundleExtractor` is not needed IMO, as extractors are not really something you inspect this way (you instantiate them to pass them to the extraction layer).
We have many cases where we create a new class and make the deprecated one extend the new one.